### PR TITLE
Treat 2pi periodicity properly in yaw tolerance decisions.

### PR DESCRIPTION
**Pull request by jasonimercer**
@p6chen @skaynama @schopra01 

Fixes wobble on goal 

CORE-4390

### CODE REVIEW

**Comment by jliviero** [comment]
Did you intend to leave in the CATKIN_IGNORE?

**Comment by jasonimercer** [comment]
Did not intend to. Thanks for spotting it.

**Comment by schopra01** [comment]
Looks good!

File: nav_core/include/nav_core/goal_tolerances.h [hunk 1]
```diff
@@ -135,8 +135,13 @@ class GoalTolerances
...
   bool lessThanTolerances(float dx, float dy, float dyaw) const
   {
+    dyaw = fmod(fmod(dyaw, 2.0*M_PI) + 2.0*M_PI, 2.0*M_PI);
+    if (dyaw > M_PI)
+    {
+      dyaw -= 2.0 * M_PI;
+    }
     return (dx * dx + dy * dy <= goal_tolerance_xy_ * goal_tolerance_xy_) &
-           (fmod(fabs(dyaw), 2.0 * M_PI) <= goal_tolerance_yaw_);
+           (dyaw <= goal_tolerance_yaw_);
```
Comment (p6chen) [inline nav_core/include/nav_core/goal_tolerances.h]: I think you need a fabs( ) here..

File: nav_core/include/nav_core/goal_tolerances.h [hunk 2]
```diff
@@ -135,8 +135,13 @@ class GoalTolerances
    */
   bool lessThanTolerances(float dx, float dy, float dyaw) const
   {
+    dyaw = fmod(fmod(dyaw, 2.0*M_PI) + 2.0*M_PI, 2.0*M_PI);
```
Comment (p6chen) [inline nav_core/include/nav_core/goal_tolerances.h]: Might be easier to just use angles::normalize_angle( )

File: nav_core/include/nav_core/goal_tolerances.h (hunk 1 above)
Comment (jasonimercer) [inline nav_core/include/nav_core/goal_tolerances.h]: :+1:

File: nav_core/include/nav_core/goal_tolerances.h (hunk 2 above)
Comment (skaynama) [inline nav_core/include/nav_core/goal_tolerances.h]: :+1:

Comment (jasonimercer) [inline nav_core/include/nav_core/goal_tolerances.h]: I agree. I just didn't want to add dependencies. I'll make a ticket to switch to angles:: once we pull navigation in.

Comment (jasonimercer) [inline nav_core/include/nav_core/goal_tolerances.h]: CORE-4435

**Comment by p6chen** [comment]
Ok lgtm

File: nav_core/include/nav_core/goal_tolerances.h (hunk 2 above)
Comment (p6chen) [inline nav_core/include/nav_core/goal_tolerances.h]: There is also the infamous: atan2(sin(dyaw), cos(dyaw) ). Slower but cleaner. Anyways up to you.

**Comment by skaynama** [comment]
LGTM

**Pull request merged**